libmscore/rendermidi.cpp

```diff
@@ -402,7 +402,10 @@ static void collectMeasureEvents(EventMap* events, Measure* m, Staff* staff, int
                               for (MidiCoreEvent event : nel->events) {
                                     event.setChannel(channel);
                                     NPlayEvent e(event);
-                                    events->insert(std::pair<int, NPlayEvent>(tick, e));
+                                    if (e.dataA() == CTRL_PROGRAM)
```

Comment:
I think it would depend. Likely in most cases it wouldn't hurt. But for legato (CC68) for example it would eventually be better if it would happen one tick after - even though I think the synth should take care of this!